hotspot/make/linux/makefiles/gcc.make

```diff
@@ -217,7 +217,19 @@ ifeq ($(USE_CLANG), true)
   WARNINGS_ARE_ERRORS += -Wno-return-type -Wno-empty-body
 endif
 
-WARNING_FLAGS = -Wpointer-arith -Wsign-compare -Wundef -Wunused-function -Wunused-value -Wformat=2 -Wreturn-type
+WARNING_FLAGS = -Wpointer-arith -Wsign-compare -Wundef -Wunused-function -Wformat=2 -Wreturn-type
+
+ifeq ($(USE_CLANG),)
+  # AL2022 has GCC 11.3.1 and also has stricter checks. Disabling until code fixes can be backported.
+  # gcc -dumpversion is only reporting "11" with no minor version.
+  ifeq "$(shell expr \( $(CC_VER_MAJOR) \>= 11 \))" "1"
+    WARNINGS_ARE_ERRORS += -Wno-char-subscripts -Wno-reorder -Wno-parentheses -Wno-unused-variable
+    WARNINGS_ARE_ERRORS += -Wno-nonnull-compare -Wno-format-overflow -Wno-maybe-uninitialized -Wno-comment
+    WARNINGS_ARE_ERRORS += -Wno-class-memaccess -Wno-unused-but-set-variable -Wno-stringop-truncation
+    WARNINGS_ARE_ERRORS += -Wno-c++11-compat -Wno-switch -Wno-delete-non-virtual-dtor -Wno-uninitialized
+    WARNINGS_ARE_ERRORS += -Wno-misleading-indentation -Wno-address
+  endif
+endif
```

Comment:
Why don't we just put those code in next section for gcc?

Reply:
Good point, moving code.